service/src/test/java/com/commercetools/util/HttpRequestUtilParallelTest.java

```diff
@@ -24,52 +26,108 @@
+     */
+    private Supplier<Integer> createRequestsSupplier(final ExceptionalSupplier<HttpResponse> responseSupplier,
+                                                     final ExceptionalConsumer<HttpResponse> responseAssertConsumer) {
+        return () -> {
+            try {
+                int statusCode;
+                int repeatedTimes = 0;
+                HttpResponse httpResponse;
+                do {
+                    if (repeatedTimes > 0) {
+                        LOGGER.warn("{} \"Request Timeout\" received from [{}]. Retry {} of {} ...",
+                                SC_REQUEST_TIMEOUT, "http://httpbin.org/", repeatedTimes, RETRY_TIMES);
+                    }
+                    httpResponse = responseSupplier.get();
+                    statusCode = httpResponse.getStatusLine().getStatusCode();
+                    repeatedTimes++;
+                    // sometimes httpbin.org returns 408 (timeout) response. Note, this is not a InterruptedIOException
+                    // or ConnectException, thus DefaultHttpRequestRetryHandler re-try won't handle this case.
+                    // That's why in the tests we use this approach to re-try if the server responded successfully,
+                    // but with "408 timeout" response
+                } while (statusCode == SC_REQUEST_TIMEOUT && repeatedTimes <= RETRY_TIMES);
+
+                assertThat(repeatedTimes)
+                        .withFailMessage(format("Retries over-limit: the post request is executed %d times with timeout response %d",
+                                repeatedTimes, SC_REQUEST_TIMEOUT))
+                        .isLessThanOrEqualTo(RETRY_TIMES);
+
+                responseAssertConsumer.accept(httpResponse);
                 return 1;
-            } catch (IOException e) {
-                throw new RuntimeException("Request exception", e);
+            } catch (Exception e) {
+                LOGGER.error("Request exception: {}", e.toString());
             }
-        });
+
+            return 0;
+        };
     }
 
-    private void makeParallelRequests(final int nThreads, final int requests, final Supplier<Integer> request) throws Exception {
+    /**
+     * Execute and count {@code requestsNumber} simultaneous (parallel) requests in {@code nThreads} generating them
+     * from {@code requestSupplier}. At the end assert that all {@code requestsNumber} finished successfully, i.e.
+     * exception or timeout not happened.
+     *
+     * @param nThreads        number of parallel requests/threads
+     * @param requestsNumber  total number of requests to execute
+     * @param requestSupplier requests generator: generated requests return <b>1</b> if finished successfully or
+     *                        <b>0</b> otherwise
+     * @throws Exception in case of thread pool exceptions
+     */
+    private void makeAndAssertParallelRequests(final int nThreads, final int requestsNumber,
```

Comment:
why `nThreads` and `requestsNumber`? 
maybe either
- `nThreads` and `nRequests` 
- `threadsNumber` and `requestsNumber`
?

Reply:
also here: https://github.com/commercetools/commercetools-payone-integration/pull/198/files#diff-68f2f161bb34687ec51123bdbb575269L28

Reply:
fc36e546